wandb/apis/public/artifacts.py

```diff
@@ -848,7 +850,15 @@ def convert_objects(self):
             for a in self.last_response["project"]["artifactType"][
                 "artifactCollection"
             ]["artifacts"]["edges"]
-        ]
+        )
+        if self.tags:
+            required_tags = set(self.tags)
+            artifacts = (
+                artifact
+                for artifact in artifacts
+                if required_tags.issubset(artifact.tags)
+            )
```

Comment:
The code reads better if you use a short list comprehension, with long portions factored out and named, and with no conditional processing:

```python
artifacts_edges = self.last_response[...]

required_tags = set(self.tags) if self.tags else set()
return [
    wandb.Artifact._from_attrs(...)
    for a in artifacts_edges
    if required_tags.issubset(artifact.tags)
]
```

Reply:
Thanks! I like it better without the conditionals